broker/cloud_run/lsst/ps_to_storage/main.py

```diff
@@ -95,148 +84,62 @@ def run():
 def store_alert_data(envelope) -> None:
     """Uploads the msg data bytes to a GCP storage bucket."""
 
-    alert_bytes = base64.b64decode(envelope["message"]["data"])  # alert packet, bytes
-    attributes = envelope["message"].get("attributes", {})
-
-    # unpack the alert and read schema ID
-    header_bytes = alert_bytes[:5]
-    schema_id = deserialize_confluent_wire_header(header_bytes)
-
-    # get and load schema
-    schema = sr_client.get_schema(schema_id=schema_id)
-    parse_schema = json.loads(schema.schema_str)
-    schema_version = parse_schema["namespace"].split(".")[1]
-    content_bytes = io.BytesIO(alert_bytes[5:])
-
-    # deserialize the alert
-    alert_dict = fastavro.schemaless_reader(content_bytes, parse_schema)
-
-    # convert the MJD timestamp to "YYYY-MM-DD"
-    time_obj = Time(alert_dict["diaSource"]["midpointMjdTai"], format="mjd")
-    alert_date = time_obj.datetime.strftime("%Y-%m-%d")
-
-    filename = generate_alert_filename(
-        {
-            "schema_version": schema_version,
-            "alert_date": alert_date,
-            "objectId": alert_dict["diaObject"]["diaObjectId"],
-            "sourceId": alert_dict["diaSource"]["diaSourceId"],
-            "format": "avro",
-        }
-    )
+    # create an alert object from the envelope
+    alert = pittgoogle.Alert.from_cloud_run(envelope, "lsst")
 
-    blob = bucket.blob(filename)
-    blob.metadata = create_file_metadata(alert_dict, event_id=envelope["message"]["messageId"])
+    blob = bucket.blob(_generate_alert_filename(alert))
+    blob.metadata = _create_file_metadata(alert, event_id=envelope["message"]["messageId"])
 
     # raise a PreconditionFailed exception if filename already exists in the bucket using "if_generation_match=0"
     # let it raise. the main function will catch it and then drop the message.
-    blob.upload_from_string(alert_bytes, if_generation_match=0)
+    blob.upload_from_string(base64.b64decode(envelope["message"]["data"]), if_generation_match=0)
+
+    json_dict = _reformat_alert_data_to_valid_json(alert)
 
-    # Cloud Storage says this is not a duplicate, so now we publish the broker's main "alerts" stream
-    publish_alerts_stream(
-        topic_name=TOPIC_ALERTS.name,
-        message=alert_bytes,
+    # publish alerts to appropriate Pub/Sub topics
+    TOPIC_ALERTS.publish(alert)  # deduplicated "alerts" stream
+    publish_valid_json_stream(
+        topic_name=TOPIC_BIGQUERY_IMPORT.name,
+        message=json_dict,
         attributes={
-            "diaObjectId": str(alert_dict["diaObject"]["diaObjectId"]),
-            "diaSourceId": str(alert_dict["diaSource"]["diaSourceId"]),
-            "schema_version": schema_version,
-            **attributes,
+            "schema_version": alert.schema.version,
```

Comment:
Should we have `pittgoogle` add this as an attribute automatically at the same time it does the ids? Does the bigquery subscription require this attribute to be called "schema_version"? Can there be other attributes included in here or will the bigquery subscription choke on them?

Reply:
> Should we have `pittgoogle` add this as an attribute automatically at the same time it does the ids? 

Yes I think that would be really useful.

> Does the bigquery subscription require this attribute to be called "schema_version"? 

As things are set up right now, yes. This can be easily changed when the subscription is created in `setup_broker.sh`.

> Can there be other attributes included in here or will the bigquery subscription choke on them?

Other attributes can be included (and are currently present in the tests I've performed). From what I've seen so far, there doesn't seem to be any issues.

Reply:
@troyraen I'm also realizing that as things are currently written in this script, the messages in the deduplicated alert stream are missing additional attributes:
- `diaObjectId`
- `diaSourceId`

Can these attributes be included as well? (if this is already being done in `pitttgoogle`, I'm not seeing where that happens)

Reply:
> > Does the bigquery subscription require this attribute to be called "schema_version"?
> As things are set up right now, yes. This can be easily changed when the subscription is created in setup_broker.sh.

Reply:
> > Does the bigquery subscription require this attribute to be called "schema_version"?

> As things are set up right now, yes. This can be easily changed when the subscription is created in setup_broker.sh.

Awesome. Let's switch the separator "_" -> "."  so that it will match what we do for nested fields -> attributes (eg, sourceid (lsst) -> "diaSource.diaSourceId"). I've added it to the `alert.attributes` in https://github.com/mwvgroup/pittgoogle-client/pull/81.

Reply:
> Can these attributes be included as well? (if this is already being done in pittgoogle, I'm not seeing where that happens)

These should be added the first time you call `alert.attributes` (which may be when the message is published). If they're not being added, that's a bug.